idaes/models/properties/modular_properties/reactions/equilibrium_forms.py

```diff
@@ -284,10 +314,12 @@ def return_expression(b, rblock, r_idx, T):
             if sunits is not None:
                 s = s / sunits
 
+        s = rblock.s_scale * s / (s + rblock.s_norm)
+
         Q = b.log_k_eq[r_idx] - e
         # Q should be unitless due to log form
 
-        return s - smooth_max(0, s - Q, rblock.eps) == 0
+        return Q - smooth_max(0, Q - s, rblock.eps) == 0
```

Comment:
If one or both Q and s need to be zero, could this just bee smooth min(Q, s) == 0?

Reply:
Current expression goes `Q-0.5(Q-s+|Q-s|) == 0` while the smooth min gives `0.5(Q+s-|Q-s|)==0`. The complexity remains the same. The current expression keeps Q as a leading term, which might be easy for users to scale the solubility product with other equilibrium reactions.

Reply:
@eslickj I think this formulation is correct - it is a bit more complicated than just one of Q or s needing to be zero.

Reply:
@andrewlee94, I'm sure you are probably right.  Can someone explain the difference to me?  Why is this one preferred?  I think both should work, but I may be missing something. To me, the the one in the code currently just seems convoluted.

Reply:
@eslickj I cannot remember the details - I asked Larry Biegler to help with this is and this is the form he suggested.